### ISSUE TITLE
CCI on Neoforge 1.20.4 causes packs to fail to load

### ISSUE THREAD

**Issue by Saereth** [iChun/ContentCreatorIntegration-IssuesAndDocumentation#96]
This error occurs on tons of mods when adding CCI to the pack on Neoforge 1.20.4
<img width="1241" alt="image" src="https://github.com/iChun/ContentCreatorIntegration-IssuesAndDocumentation/assets/9200909/1459ff51-d7c1-4cf5-84e9-1e7000a8689b">

The world fails to load fully with no value present error
full logfile here:
https://mclo.gs/JR9kDn3

ContentCreatorIntegration-1.20.4-NeoForge-1.11.1.jar
Neoforge: 20.4.234
mc: 1.20.4

Error does not occur and world loads fine with CCI present

**Comment by iChun** [comment]
In chats with Joprebond about this, not entirely sure what's causing it since there's no glaringly obvious errors thrown by CCI. Will try and look into in the coming week

**Comment by TheonlyTazz** [comment]
actual error i found is (for what its worth):
```
[09Jun2024 11:16:24.344] [Netty Local Client IO #1/ERROR] [net.minecraft.network.Connection/]: Exception caught in connection
java.util.NoSuchElementException: No value present
        at java.util.Optional.get(Optional.java:143) ~[?:?]
        at me.ichun.mods.cci.common.network.PacketChannel$PacketPayload.process(PacketChannel.java:120) ~[ContentCreatorIntegration-1.20.4-NeoForge-1.11.1.jar%23185!/:1.11.1]
        at net.neoforged.neoforge.network.registration.PlayRegistration.handle(PlayRegistration.java:36) ~[neoforge-20.4.234-universal.jar%23192!/:?]
        at net.neoforged.neoforge.network.registration.NetworkRegistry.onModdedPacketAtClient(NetworkRegistry.java:432) ~[neoforge-20.4.234-universal.jar%23192!/:?]
        at net.minecraft.client.multiplayer.ClientPacketListener.handleCustomPayload(ClientPacketListener.java:1936) ~[client-1.20.4-20231207.154220-srg.jar%23191!/:?]
        at net.minecraft.client.multiplayer.ClientCommonPacketListenerImpl.handleCustomPayload(ClientCommonPacketListenerImpl.java:104) ~[client-1.20.4-20231207.154220-srg.jar%23191!/:?]
        at net.minecraft.network.protocol.common.ClientboundCustomPayloadPacket.handle(ClientboundCustomPayloadPacket.java:123) ~[client-1.20.4-20231207.154220-srg.jar%23191!/:?]
        at net.minecraft.network.protocol.common.ClientboundCustomPayloadPacket.handle(ClientboundCustomPayloadPacket.java:30) ~[client-1.20.4-20231207.154220-srg.jar%23191!/:?]
        at net.minecraft.network.Connection.genericsFtw(Connection.java:194) ~[client-1.20.4-20231207.154220-srg.jar%23191!/:?]
        at net.minecraft.network.Connection.channelRead0(Connection.java:178) ~[client-1.20.4-20231207.154220-srg.jar%23191!/:?]
        at net.minecraft.network.Connection.channelRead0(Connection.java:60) ~[client-1.20.4-20231207.154220-srg.jar%23191!/:?]
        at io.netty.channel.SimpleChannelInboundHandler.channelRead(SimpleChannelInboundHandler.java:99) ~[netty-transport-4.1.97.Final.jar%23153!/:4.1.97.Final]
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:444) ~[netty-transport-4.1.97.Final.jar%23153!/:4.1.97.Final]
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420) ~[netty-transport-4.1.97.Final.jar%23153!/:4.1.97.Final]
        at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412) ~[netty-transport-4.1.97.Final.jar%23153!/:4.1.97.Final]
        at io.netty.handler.flow.FlowControlHandler.dequeue(FlowControlHandler.java:202) ~[netty-handler-4.1.97.Final.jar%23149!/:4.1.97.Final]
        at io.netty.handler.flow.FlowControlHandler.channelRead(FlowControlHandler.java:164) ~[netty-handler-4.1.97.Final.jar%23149!/:4.1.97.Final]
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:442) ~[netty-transport-4.1.97.Final.jar%23153!/:4.1.97.Final]
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420) ~[netty-transport-4.1.97.Final.jar%23153!/:4.1.97.Final]
        at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412) ~[netty-transport-4.1.97.Final.jar%23153!/:4.1.97.Final]
        at io.netty.handler.codec.MessageToMessageDecoder.channelRead(MessageToMessageDecoder.java:103) ~[netty-codec-4.1.97.Final.jar%23147!/:4.1.97.Final]
        at io.netty.handler.codec.MessageToMessageCodec.channelRead(MessageToMessageCodec.java:111) ~[netty-codec-4.1.97.Final.jar%23147!/:4.1.97.Final]
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:442) ~[netty-transport-4.1.97.Final.jar%23153!/:4.1.97.Final]
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420) ~[netty-transport-4.1.97.Final.jar%23153!/:4.1.97.Final]
        at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412) ~[netty-transport-4.1.97.Final.jar%23153!/:4.1.97.Final]
        at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1410) ~[netty-transport-4.1.97.Final.jar%23153!/:4.1.97.Final]
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:440) ~[netty-transport-4.1.97.Final.jar%23153!/:4.1.97.Final]
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420) ~[netty-transport-4.1.97.Final.jar%23153!/:4.1.97.Final]
        at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:919) ~[netty-transport-4.1.97.Final.jar%23153!/:4.1.97.Final]
        at io.netty.channel.local.LocalChannel.readInbound(LocalChannel.java:299) ~[netty-transport-4.1.97.Final.jar%23153!/:4.1.97.Final]
        at io.netty.channel.local.LocalChannel.doBeginRead(LocalChannel.java:322) ~[netty-transport-4.1.97.Final.jar%23153!/:4.1.97.Final]
        at io.netty.channel.AbstractChannel$AbstractUnsafe.beginRead(AbstractChannel.java:834) ~[netty-transport-4.1.97.Final.jar%23153!/:4.1.97.Final]
        at io.netty.channel.DefaultChannelPipeline$HeadContext.read(DefaultChannelPipeline.java:1362) ~[netty-transport-4.1.97.Final.jar%23153!/:4.1.97.Final]
        at io.netty.channel.AbstractChannelHandlerContext.invokeRead(AbstractChannelHandlerContext.java:835) ~[netty-transport-4.1.97.Final.jar%23153!/:4.1.97.Final]
        at io.netty.channel.AbstractChannelHandlerContext.read(AbstractChannelHandlerContext.java:814) ~[netty-transport-4.1.97.Final.jar%23153!/:4.1.97.Final]
        at io.netty.handler.flow.FlowControlHandler.read(FlowControlHandler.java:144) ~[netty-handler-4.1.97.Final.jar%23149!/:4.1.97.Final]
        at io.netty.channel.AbstractChannelHandlerContext.invokeRead(AbstractChannelHandlerContext.java:837) ~[netty-transport-4.1.97.Final.jar%23153!/:4.1.97.Final]
        at io.netty.channel.AbstractChannelHandlerContext.read(AbstractChannelHandlerContext.java:814) ~[netty-transport-4.1.97.Final.jar%23153!/:4.1.97.Final]
        at io.netty.channel.DefaultChannelPipeline.read(DefaultChannelPipeline.java:1004) ~[netty-transport-4.1.97.Final.jar%23153!/:4.1.97.Final]
        at io.netty.channel.AbstractChannel.read(AbstractChannel.java:290) ~[netty-transport-4.1.97.Final.jar%23153!/:4.1.97.Final]
        at io.netty.channel.DefaultChannelPipeline$HeadContext.readIfIsAutoRead(DefaultChannelPipeline.java:1422) ~[netty-transport-4.1.97.Final.jar%23153!/:4.1.97.Final]
        at io.netty.channel.DefaultChannelPipeline$HeadContext.channelReadComplete(DefaultChannelPipeline.java:1417) ~[netty-transport-4.1.97.Final.jar%23153!/:4.1.97.Final]
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelReadComplete(AbstractChannelHandlerContext.java:482) ~[netty-transport-4.1.97.Final.jar%23153!/:4.1.97.Final]
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelReadComplete(AbstractChannelHandlerContext.java:463) ~[netty-transport-4.1.97.Final.jar%23153!/:4.1.97.Final]
        at io.netty.channel.DefaultChannelPipeline.fireChannelReadComplete(DefaultChannelPipeline.java:925) ~[netty-transport-4.1.97.Final.jar%23153!/:4.1.97.Final]
        at io.netty.channel.local.LocalChannel.readInbound(LocalChannel.java:302) ~[netty-transport-4.1.97.Final.jar%23153!/:4.1.97.Final]
        at io.netty.channel.local.LocalChannel.finishPeerRead0(LocalChannel.java:445) ~[netty-transport-4.1.97.Final.jar%23153!/:4.1.97.Final]
        at io.netty.channel.local.LocalChannel.access$400(LocalChannel.java:50) ~[netty-transport-4.1.97.Final.jar%23153!/:4.1.97.Final]
        at io.netty.channel.local.LocalChannel$5.run(LocalChannel.java:403) ~[netty-transport-4.1.97.Final.jar%23153!/:4.1.97.Final]
        at io.netty.util.concurrent.AbstractEventExecutor.runTask(AbstractEventExecutor.java:174) ~[netty-common-4.1.97.Final.jar%23148!/:4.1.97.Final]
        at io.netty.channel.DefaultEventLoop.run(DefaultEventLoop.java:54) ~[netty-transport-4.1.97.Final.jar%23153!/:4.1.97.Final]
        at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:997) ~[netty-common-4.1.97.Final.jar%23148!/:4.1.97.Final]
        at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74) ~[netty-common-4.1.97.Final.jar%23148!/:4.1.97.Final]
        at java.lang.Thread.run(Thread.java:833) ~[?:?]
```

can be replicated with just CCI on Neoforge

**Comment by iChun** [comment]
That helps plenty, thanks

**Comment by iChun** [comment]
Should be fixed in v1.11.2
